network/gossip/libp2p/libp2pNode.go

```diff
@@ -526,6 +526,17 @@ func (p *P2PNode) UpdateAllowlist(allowListAddrs ...NodeAddress) error {
 	return nil
 }
 
+// IsConnected returns true is address is a direct peer of this node else false
+func (p *P2PNode) IsConnected(address NodeAddress) (bool, error) {
+	pInfo, err := GetPeerInfo(address)
+	if err != nil {
+		return false, err
+	}
+	// query libp2p for connectedness status of this peer
+	isConnected := p.libP2PHost.Network().Connectedness(pInfo.ID) == network.Connected
+	return isConnected, nil
```

Comment:
maybe do this: `return p.libP2PHost.Network().Connectedness(pInfo.ID) == network.Connected, nil`

Reply:
I think both of us think alike - I had exactly that way and Jordan suggested adding a new variable `isConnected`. I think, the additional variable makes the code a little more readable.